### ISSUE TITLE
Umbrella header 'CorePlot.h' not found

### ISSUE THREAD

**Issue by sprout-apps** [core-plot/core-plot#477]
Today my project stopped building due to the following error:

core-plot/framework/Module/module.modulemap:2:19: Umbrella header 'CorePlot.h' not found

MacOS: 13.2.1 (22D68)
Xcode: Version 14.3 (14E222b)

**Comment by eskroch** [comment]
How are you integrating Core Plot into your app? CocoaPods? Building from source from GitHub? Make sure you're using the latest code from the `release-2.4` branch.

**Comment by sprout-apps** [comment]
Yes, l'm using the latest release-2.4 branch and building from source. core-plot has been integrated into our projects for close to 10 years. Both are mixed projects with Objective-C and Swift and both are failing to build with the same error.

**Comment by sprout-apps** [comment]
I was wondering if there was any update on this issue? Without the fix we are not able to make any development or production builds as our apps are heavily dependent on core-plot.

**Comment by eskroch** [comment]
I suspect there's a project setting in your app that conflicts with recent updates to Core Plot done to fix various issues with the module interface.

Is there a way you could share your project or a stripped-down version with similar settings to help troubleshoot?

**Comment by sprout-apps** [comment]
Unfortunately, I can't share the project code at this time. However, I do have more information. The last build that works is e0b8dd16. Here is some additional info from our developer:

"After that build there are various issues: with module verifier error; with double quoted includes error; and last one with a module map error."

He also believes that creating an empty project in Xcode with iOS app and importing core-plot source code directly into it would produce this error. I haven't tried it yet.

**Comment by myoung959** [comment]
Same problem here. I've been rebuilding all my apps that use CorePlot with the release-2.4 branch. Three worked fine, but the fourth fails and fails, even after rebuilding from scratch always with 'CorePlot.h" not found in file xxx.m even though #import "CorePlot.h" is in file xxx.h.

**Comment by myoung959** [comment]
Yes, I'll be happy to send you my source code.

**Comment by eskroch** [comment]
> Yes, I'll be happy to send you my source code.

My email address is listed in my profile. Thanks!

**Comment by eskroch** [comment]
I believe the build problems have been resolved on the `release-2.4` branch. Please let me know if you're still seeing any issues.

**Comment by sprout-apps** [comment]
Thanks for making the changes. I just tested and unfortunately I'm still seeing the same issue:

Umbrella header 'CorePlot.h' not found
Could not build Objective-C module 'CorePlot'

**Comment by eskroch** [comment]
Does the error occur when building Core Plot by itself or just when building your app?

**Comment by sprout-apps** [comment]
No, I can build Core Plot by itself, so only when building the app. It's a mix of Objective-C and Swift.

**Comment by eskroch** [comment]
That sounds like an issue with the build configuration of your app. The Mac `CPTTestApp` example app (Objective-C) uses module references and two of the example apps (`DatePlot` and the iPhone `CPTTestApp`) are written in Swift. Pull the latest code from the `release-2.4` branch and see if you can build those apps.

Take a look at the CorePlot framework in your build folder. The `Headers` folder inside the framework should contain 68 files, including `CorePlot.h` and 67 with filenames beginning with `CPT*`.

**Comment by sprout-apps** [comment]
Unfortunately we are facing a deadline and I can't dedicate the time to follow your suggestions. For now we will continue to build using the last version that works: e0b8dd16. I'll try debugging on my end after our release.
